oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticConnection.java

```diff
@@ -108,11 +123,41 @@ public RestHighLevelClient getClient() {
                         Header[] headers = new Header[]{new BasicHeader("Authorization", "ApiKey " + apiKeyAuth)};
                         builder.setDefaultHeaders(headers);
                     }
-                    client = new RestHighLevelClient(builder);
+                    RestClient httpClient = builder.build();
+                    hlClient = new RestHighLevelClientBuilder(httpClient).setApiCompatibilityMode(true).build();
+
+                    ElasticsearchTransport transport = new RestClientTransport(
+                            httpClient, new JacksonJsonpMapper());
+                    esClient = new ElasticsearchClient(transport);
+                    esAsyncClient = new ElasticsearchAsyncClient(transport);
                 }
             }
         }
-        return client;
+        return Tuple.tuple(Tuple.tuple(esClient, esAsyncClient), hlClient);
+    }
+
+    /**
+     * Gets the Elasticsearch Client
+     * @return the Elasticsearch client
+     */
+    public ElasticsearchClient getClient() {
+        return getClients().v1().v1();
```

Comment:
Makes sense. I have created an internal class (`Clients`) to wrap up all the clients. I preferred it to a Map since it's strongly typed (no need to cast).